custom_components/waste_collection_schedule/waste_collection_schedule/source/aw_harburg.py

```diff
@@ -0,0 +1,203 @@
+                    )
+                )
+            soup = BeautifulSoup(response.text, features="html.parser")
+            select_content = soup.find_all("select", id="strukturEbene3")
+            soup = BeautifulSoup(str(select_content), features="html.parser")
+            options_content = soup.find_all("option")
+            level_3_ids = {}
+            for option in options_content:
+                # Ignore the "Bitte wählen..."
+                if option.get("value")!="0":
+                    level_3_ids[option.text] = option.get("value")
+            if level_3_ids == {}:
+                raise Exception(
+                    "Error: Level 3 Dictionary empty"
+                    )
+            if self._district_level_3 not in level_3_ids:
+                raise Exception(
+                    "Error: District 3 is not in the dictionary: {}".format(
+                        (self._district_level_3, level_3_ids)
+                    )
+                )
+
+        # Prepare data for the real web request
+        if self._district_level_3 != None:
+            url = 'https://www.landkreis-harburg.de/abfallkalender/abfallkalender_struktur_daten_suche.html?selected_ebene=' + level_3_ids[self._district_level_3] + '&owner=20100'
+        else:
+            url = 'https://www.landkreis-harburg.de/abfallkalender/abfallkalender_struktur_daten_suche.html?selected_ebene=' + level_2_ids[self._district_level_2] + '&owner=20100'
+
+        response = s.get(url, headers=headers)
+        # Sometimes there is no garbage calendar available
+        if "Es sind keine Abfuhrbezirke hinterlegt." in response.text:
+            raise Exception(
+                "Error: \"Es sind keine Abfuhrbezirke hinterlegt.\" for \"" + self._district_level_3 + "\" please use different input data."
+            )
+        soup = BeautifulSoup(response.text, features="html.parser")
+        links = soup.find_all("a")
+        ical_url = ""
+        for any_link in links:
+            if " als iCal" in any_link.text:
+                ical_url = any_link.get("href")
+
+        if "ical.html" not in ical_url:
+            raise Exception(
+                "No ical Link in the result: " + str(links)
+            )
+
+        # Get the final data
+        response = s.post(ical_url, headers=headers)
+
+        # Stop if something else as status code 200 is returned
+        if response.status_code != 200:
+            raise Exception(
+                "Error: failed to fetch ical_url: {}".format(
+                    ical_url
+                )
+            )
+
+        return self.fetch_ics(ical_url, headers=headers)
+
+    def fetch_ics(self, url, headers={}):
+        r = requests.get(url, headers=headers)
+
+        if not r.ok:
+            raise Exception(
+                "Error: failed to fetch url: {}".format(
+                    url
+                )
+            )
+        
+        # Parse ics file, fix broken encoding
+        if r.encoding=="ISO-8859-1":
```

Comment:
Would `r.encoding = "utf-8"` also work? (Instead of the if else). I used this many times for other sources.

Reply:
I tried it without that, didn't work. For some reason the webservice or the python requests module received ISO-8859-1 encoded data.